From bfd6bf03805103c4d1af9f884308a7faae3dbea4 Mon Sep 17 00:00:00 2001
From: Colin Cross <ccross@android.com>
Date: Thu, 12 Apr 2012 14:55:21 -0700
Subject: [PATCH 180/944] android: ram_console: use platform_device_register

platform_driver_probe can only be used when the platform device
is already present.  In order to enable debugging crashes during
very early boot, the ram_console driver needs to be registered
as early as possible to allow the device to probe as soon as it
is registered at arch_initcall time.  This requires the driver
to be registered before the device.

Replace platform_driver_probe with platform_driver_register,
and make the ram_console probe function __devinit.

Change-Id: Ie02cebef90b9c9dbb5b94e52e4f93e490bb88954
Signed-off-by: Colin Cross <ccross@android.com>
---
 drivers/staging/android/ram_console.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/android/ram_console.c b/drivers/staging/android/ram_console.c
index 047f193..8e4038b 100644
--- a/drivers/staging/android/ram_console.c
+++ b/drivers/staging/android/ram_console.c
@@ -50,7 +50,7 @@ void ram_console_enable_console(int enabled)
 		ram_console.flags &= ~CON_ENABLED;
 }
 
-static int __init ram_console_probe(struct platform_device *pdev)
+static int __devinit ram_console_probe(struct platform_device *pdev)
 {
 	struct ram_console_platform_data *pdata = pdev->dev.platform_data;
 	struct persistent_ram_zone *prz;
@@ -78,11 +78,12 @@ static struct platform_driver ram_console_driver = {
 	.driver		= {
 		.name	= "ram_console",
 	},
+	.probe = ram_console_probe,
 };
 
 static int __init ram_console_module_init(void)
 {
-	return platform_driver_probe(&ram_console_driver, ram_console_probe);
+	return platform_driver_register(&ram_console_driver);
 }
 
 #ifndef CONFIG_PRINTK
-- 
1.8.0

